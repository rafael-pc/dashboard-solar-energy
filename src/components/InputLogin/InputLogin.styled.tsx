import styled from "styled-components";
import { theme } from "../../styles";

export const Content = styled.div`
  width: 100%;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 3px;
  margin-top: 15px;
  font-size: ${theme.font.sizes.xxxxs}; 
  border: 1px solid ${theme.colors.gray};
  border-radius: ${theme.border.radius.s};

  &.error {
    border-color: ${theme.colors.error};
  }

  &:hover, &:focus-within {
    border: 2px solid ${theme.colors.bg_primary};
  }

  .icon {
    width: 20px;
    height: 20px;
    margin-left: 4px;
    color: ${theme.colors.gray};
  }

  @media only screen and (max-height: ${theme.screen.mobile}) {
    margin-top: 10px;
  }
`;

export const Input = styled.input`
  width: 100%;
  flex: 1 0;
  font-size: ${theme.font.sizes.xxxs};
  background-color: transparent;
  margin-left: 6px;
  padding: 7px;
  border: 0;
  border-radius: ${theme.border.radius.s};
  background-color: ${theme.colors.neutral_100};
`;
