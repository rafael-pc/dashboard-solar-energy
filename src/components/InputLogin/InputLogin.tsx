import React, { InputHTMLAttributes, ReactNode } from "react";
import { Input, Content } from "./InputLogin.styled";

interface InputLoginProps extends InputHTMLAttributes<HTMLInputElement> {
  icon?: ReactNode;
}

const InputLogin: React.FC<InputLoginProps> = ({icon, ...props}) => {
  return (
    <>
      <Content>
        {icon}
        <Input {...props}/>
      </Content>
    </>
  );
};

export default InputLogin;
