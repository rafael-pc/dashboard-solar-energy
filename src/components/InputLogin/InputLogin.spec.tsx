import { render, screen } from '@testing-library/react';
import InputLogin from './InputLogin';
import "@testing-library/jest-dom";

describe('InputLogin', () => {
  it('renders input and icon', () => {
    const Icon = () => <i data-testid="icon">icon</i>;
    const { asFragment } = render(<InputLogin icon={<Icon />} />);
    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByTestId('icon')).toBeInTheDocument();
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });
});
