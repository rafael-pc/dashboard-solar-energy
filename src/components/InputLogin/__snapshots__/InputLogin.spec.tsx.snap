// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLogin renders input and icon 1`] = `
<DocumentFragment>
  <div
    class="sc-bcPKhP gZVulA"
  >
    <i
      data-testid="icon"
    >
      icon
    </i>
    <input
      class="sc-grXZZQ cwBNSp"
    />
  </div>
</DocumentFragment>
`;
